src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -482,15 +482,14 @@ internal override void ForceComplete(SourceLocation locationOpt, CancellationTok
                         foreach (var typeParameter in this.TypeParameters)
                         {
                             typeParameter.ForceComplete(locationOpt, cancellationToken);
-                            var diagnostics = DiagnosticBag.GetInstance();
 
                             if (typeParameter.HasUnmanagedTypeConstraint)
```

Comment:
>if (typeParameter.HasUnmanagedTypeConstraint) [](start = 28, length = 45)

This if belongs in typeParameter.ForceComplete

Reply:
**Consolidating for other similar comments**
There are three places this check is made:
* member containers: modifyCompilationForIsUnmanaged is true
* methods: modifyCompilationForIsUnmanaged is true
* local functions: modifyCompilationForIsUnmanaged is false
This looks better to me over checking in the type parameter symbol whether the containing symbol is a local function, or a method, or a type.

---
In reply to: [177238461](https://github.com/dotnet/roslyn/pull/25680#discussion_r177238461) [](ancestors = 177238461)

Reply:
> Please let me know if you still see otherwise.

1. These checks belong in completion, this is where other aspects of constraints are validated.
2. Duplicating the same code in three different places because they differ by a boolean value that can easily be calculated by looking at the containing symbol is definitely not better.

---
In reply to: [177267287](https://github.com/dotnet/roslyn/pull/25680#discussion_r177267287) [](ancestors = 177267287,177238461)

Reply:
Also, with the current implementation, how can we be sure that you are not missing a forth place where the code should be duplicated?

---
In reply to: [177269556](https://github.com/dotnet/roslyn/pull/25680#discussion_r177269556) [](ancestors = 177269556,177267287,177238461)

Reply:
If later a new symbol was introduced that can hold type parameters, it would need to make that decision about whether to modify the compilation or not.
I'll move to the TypeParamterSymbol and throw otherwise.

---
In reply to: [177270545](https://github.com/dotnet/roslyn/pull/25680#discussion_r177270545) [](ancestors = 177270545,177269556,177267287,177238461)